aws/modules/vpc/main.tf

```diff
@@ -6,11 +10,34 @@ resource "aws_vpc" "ts_aws_vpc" {
 
 }
 
+locals {
+
+    available_azs = var.ts_vpc_slice_azs ? slice(data.aws_availability_zones.available.names, var.ts_vpc_slice_azs_start_index, var.ts_vpc_slice_azs_end_index) : data.aws_availability_zones.available.names
```

Comment:
The customer is not going to know the indexes of a particular az; they might not even know how many az's are in a given region.  We should sort the az's by name so that they're always in the same order, then always start at index 0 and accept a `maxAzs` arg that defaults to 3.  Then we can choose between this max or the count of az's if it is less than the max.